edit-event/DurationPicker.tsx

```diff
@@ -1,9 +1,399 @@
+} from "react-native"
+import { Gesture, GestureDetector } from "react-native-gesture-handler"
+import Animated, {
+  runOnJS,
+  SharedValue,
+  useAnimatedStyle,
+  useSharedValue
+} from "react-native-reanimated"
+import { formatEventDurationPreset } from "TiFShared/domain-models/Settings"
+
+const INNER_TRACK_GAP = 8
 
 export type EditEventDurationPickerProps = {
+  // NB: An atom gives us a significant performance boost over using a set of
+  // value/onValueChange props.
+  durationAtom: PrimitiveAtom<number>
+  presetOptions: number[]
   style?: StyleProp<ViewStyle>
 }
 
 export const EditEventDurationPickerView = ({
+  durationAtom,
+  presetOptions,
   style
-}: EditEventDurationPickerProps) => <></>
+}: EditEventDurationPickerProps) => {
+  const setDuration = useSetAtom(durationAtom)
+  const [pickerState, setPickerState] = useState(
+    initialPickerState(presetOptions)
+  )
+  const isSliding = useSharedValue(false)
+  const sliderPosition = useSharedValue(0)
+  const height =
+    80 * useFontScale({ maximumScaleFactor: FontScaleFactors.large })
+  useEffect(
+    () => setPickerState(initialPickerState(presetOptions)),
+    [presetOptions]
+  )
+  return (
+    <View style={style}>
+      <View style={styles.container}>
+        <View style={[styles.backgroundCard, { height }]} />
+        <BackgroundTrailView
+          height={height}
+          isSliding={isSliding}
+          sliderPosition={sliderPosition}
+        />
+        <DurationsRowView
+          durations={Array.from(pickerState.keys())}
+          height={height}
+          onDurationLayout={(duration, layout) => {
+            setPickerState((state) => updateLayout(state, duration, layout))
+          }}
+          onDurationChange={setDuration}
+        />
+        <SliderKnobView
+          durationAtom={durationAtom}
+          isSliding={isSliding}
+          sliderPosition={sliderPosition}
+          pickerState={pickerState}
+          height={height}
+        />
+      </View>
+    </View>
+  )
+}
+
+type BackgroundTrailProps = {
+  sliderPosition: SharedValue<number>
+  isSliding: SharedValue<boolean>
+  height: number
+}
+
+const BackgroundTrailView = ({
+  sliderPosition,
+  isSliding,
+  height
+}: BackgroundTrailProps) => (
+  <Animated.View
+    style={[
+      styles.backgroundTrail,
+      useAnimatedStyle(() => ({
+        width: sliderPosition.value + INNER_TRACK_GAP * 3,
+        height: withTiFDefaultSpring(
+          height - (isSliding.value ? 0 : INNER_TRACK_GAP * 2)
+        ) as number,
+        marginTop: withTiFDefaultSpring(!isSliding.value ? INNER_TRACK_GAP : 0),
+        marginLeft: withTiFDefaultSpring(
+          !isSliding.value ? INNER_TRACK_GAP : 0
+        ),
+        borderBottomLeftRadius: withTiFDefaultSpring(
+          isSliding.value ? INNER_TRACK_GAP * 2 : INNER_TRACK_GAP
+        ),
+        borderTopLeftRadius: withTiFDefaultSpring(
+          isSliding.value ? INNER_TRACK_GAP * 2 : INNER_TRACK_GAP
+        )
```

Comment:
So with these, since they use `INNER_TRACK_GAP` often, why was that used instead of having the plain results from multiplication?
Would it be for versatility, since `INNER_TRACK_GAP` could be changed at some point?

Reply:
Yes

But also because it wasn't obvious _why_ I had to multiply by the padding if I just made it a magic number. If you look at the code for some of the functions that use `EditEventDurationPickerState`, you'll also see use of the content. By naming this constant, it's obvious that the padding refers to the space between the inner track (The inner black background) and the outer track.